DuckDuckGoTests/Subscription/StorePurchaseManagerTests.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  StorePurchaseManagerTests.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+@testable import Subscription
+import SubscriptionTestingUtilities
+import StoreKitTest
+
+final class StorePurchaseManagerTests: XCTestCase {
+
+    private struct Constants {
+        static let externalID = UUID().uuidString
+        static let monthlySubscriptionID = "ios.subscription.1month"
+        static let yearlySubscriptionID = "ios.subscription.1year"
+    }
+
+    var session: SKTestSession!
+    var storePurchaseManager: StorePurchaseManager!
+
+    override func setUpWithError() throws {
+        let path = Bundle.main.url(forResource: "Subscription", withExtension: "storekit")
+
+        session = try SKTestSession(contentsOf: path!)
+        session.resetToDefaultState()
+        session.disableDialogs = true
+        session.clearTransactions()
+
+        storePurchaseManager = DefaultStorePurchaseManager()
+    }
+
+    override func tearDownWithError() throws {
+        storePurchaseManager = nil
+        session = nil
+    }
+
+    func testSubscriptionOptionsWhenNoCachedProducts() async throws {
+        let subscriptionOptions = await storePurchaseManager.subscriptionOptions()
```

Comment:
Minor: Do we not annotate tests on iOS with `GIVEN-WHEN-THEN`?

Reply:
Actually, I see. you annotate `SubscriptionPagesUseSubscriptionFeatureTests`, so maybe just an oversight here.

Reply:
Will update that.